raven/raven.go

```diff
@@ -268,3 +306,29 @@ func (T *transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	defer timer.Stop()
 	return T.httpTransport.RoundTrip(req)
 }
+
+type EventEncoder interface {
```

Comment:
Also another way we could do this in tests which wouldn't require this encoder machinery is to use net/http/httptest.Server and just check what the server receives.